lib/holidays/finder/rules/year_range.rb

```diff
@@ -0,0 +1,82 @@
+# Please note that only one condition needs to match in order for `call` to return `true.
+# See the test file for this class for specific examples.
+module Holidays
+  module Finder
+    module Rules
+      class YearRange
+        class << self
+          BEFORE = :before
+          AFTER = :after
+          LIMITED = :limited
+          BETWEEN = :between
+
+          #TODO Can we just accept symbols here? Why accept strings?
```

Comment:
Right! I totally see what you are saying. I ran into a similar error, which is related to my comment above. The ranges were coming in as strings. Reading from YAML does not automatically convert it to whatever the corresponding Ruby thing is.

I'm fine leaving this as-is and removing the comment. I'll try to do it this week. Thanks for explaining.